packages/astro/src/types/public/integrations.ts

```diff
@@ -225,13 +217,18 @@ export interface BaseIntegrationHooks {
 	'astro:build:done': (options: {
 		pages: { pathname: string }[];
 		dir: URL;
+		/** @deprecated Use `routes` from `astro:routes:resolved` instead */
```

Comment:
If we intend to remove this, we'll need to think more about how to provide the existing functionality elsewhere. Fields not exposed on the new hook are relied upon by integrations and adapters, including official ones:
- Sitemap [relies on `type`, `pathname`, and `generate`](https://github.com/withastro/astro/blob/75a84dce5533ceb9bc9792d462272c453c0890d8/packages/integrations/sitemap/src/index.ts#L106-L119)
- Most adapters [rely](https://github.com/withastro/adapters/blob/73cba1db1794e9079edaec00950d6d71ecc0f116/packages/cloudflare/src/index.ts#L359-L368) on [`type`](https://github.com/withastro/adapters/blob/73cba1db1794e9079edaec00950d6d71ecc0f116/packages/vercel/src/lib/redirects.ts#L129-L134) to [emit redirects](https://github.com/withastro/adapters/blob/73cba1db1794e9079edaec00950d6d71ecc0f116/packages/netlify/src/index.ts#L217-L237) efficiently in their specific format
- The Cloudflare adapter [relies on the `segments`](https://github.com/withastro/adapters/blob/73cba1db1794e9079edaec00950d6d71ecc0f116/packages/cloudflare/src/utils/generate-routes-json.ts#L225-L226)

Reply:
Indeed! Hence the need to think about all the properties we should include, it doesn't need to be a 1:1

Reply:
`distURL` **shouldn't** be part of the new type, however it should be provided in the hook `build:done`;

Perhaps, it makes sense to evaluate a new type for the `build:done` hook, and **not** pass certain types, such as `params`, `segments`, `generate`.

Reply:
The only field used on official integrations/adapters that can't be recomputed on the consumer is `type` to indicate whether the route is an endpoint, page, redirect, or fallback.

- `pathname` is now `pattern`, so it is just a rename
- `segments` can be computed from `pattern`
- `generate` can be created from `segments`

Reply:
Then we can deprecate properties on `build:done` `routes`, instead of the whole `routes` list

Reply:
> Then we can deprecate properties on `build:done` `routes`, instead of the whole `routes` list

Yeah that sounds like a good thing